cors.go

```diff
@@ -132,15 +163,42 @@ func (cors *Cors) handlePreflight(w http.ResponseWriter, r *http.Request) {
 	options := cors.options
 	headers := w.Header()
 	origin := r.Header.Get("Origin")
-	if r.Method != "OPTIONS" || origin == "" || !cors.isOriginAllowed(origin) {
+
+	if r.Method != "OPTIONS" {
+		if cors.options.Debug {
+			cors.options.log.Printf("  Preflight aborted: %s!=OPTIONS", r.Method)
+		}
 		return
 	}
+
+	if origin == "" {
+		if cors.options.Debug {
+			cors.options.log.Printf("  Preflight aborted: empty origin")
+		}
+		return
+	}
+
+	if !cors.isOriginAllowed(origin) {
+		if cors.options.Debug {
+			cors.options.log.Printf("  Preflight aborted: origin '%s' not allowed", origin)
+		}
+		return
+	}
+
 	reqMethod := r.Header.Get("Access-Control-Request-Method")
 	if !cors.isMethodAllowed(reqMethod) {
+		if cors.options.Debug {
+			cors.options.log.Printf("  Preflight aborted: method '%s' not allowed", reqMethod)
+		}
+
 		return
 	}
 	reqHeaders := parseHeaderList(r.Header.Get("Access-Control-Request-Headers"))
 	if !cors.areHeadersAllowed(reqHeaders) {
+		if cors.options.Debug {
+			cors.options.log.Printf("  Preflight aborted: headers '%v' not allowed", reqHeaders)
+		}
+
```

Comment:
I actually did this deliberately to make information about each request easily distinguishable.  The idea is that indentation structure parallels the call tree.  But I can remove it, no problem.